Sources/PathKit.swift

```diff
@@ -152,8 +152,18 @@ extension Path {
   ///
   public func abbreviate() -> Path {
 #if os(Linux)
-    // TODO: actually de-normalize the path
-    return self
+    let env = ProcessInfo.processInfo.environment
+    guard let home = env["HOME"], !home.isEmpty else { return self }
+    guard let homeRange = self.path.range(of: home, options: [.anchored, .caseInsensitive]) else { return self }
+    let withoutHome = Path(self.path.replacingCharacters(in: homeRange, with: ""))
+    
+    if withoutHome.path.isEmpty || withoutHome.path == "/" {
+        return Path("~")
+    } else if withoutHome.isAbsolute {
+        return Path("~" + withoutHome.path)
+    } else {
+        return Path("~/" + withoutHome.path)
```

Comment:
Could use `Path("~") + withoutHome` too so you don't have to worry about `Path.separator`